core/rawdb/accessors_chain.go

```diff
@@ -419,3 +419,22 @@ func FindCommonAncestor(db DatabaseReader, a, b *block.Header) *block.Header {
 	}
 	return a
 }
+
+func IteratorBlocks(iterator DatabaseIterator, cb func(blockNum uint64, hash common.Hash) bool) {
+	iter := iterator.NewIteratorWithPrefix(headerPrefix)
+	defer iter.Release()
+
+	for iter.Next() {
```

Comment:
It will not iterating through all blocks, Every time we prune beacon chain data within a limited time(currently it's 200ms).

See https://github.com/harmony-one/harmony/pull/3908/files#diff-53d5f4b8a536ec2a8c8c92bf70b8268f1d77ad77e9f316e6f68a2bcae5303215R1347 and https://github.com/harmony-one/harmony/pull/3908/files#diff-53d5f4b8a536ec2a8c8c92bf70b8268f1d77ad77e9f316e6f68a2bcae5303215R1372

Reply:
currently the pruning operator is not blocking

Reply:
I don't quite understand. What's the range that this iterator is iterating on?